### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by facucompe**
## Summary

- Authentication for some endpoints

## Trello Card
https://trello.com/c/Xow23gXJ/30-integrar-spring-security

### CODE REVIEW

**Review by zeminlu — CHANGES_REQUESTED** [review]
Looks great!
Only one thing missing... password should be updated in a sepparate endpoint. It's the last line of the card description ;)
You should never update password on the common update endpoint you have here, but rather in a sepparate one.

**Review by zeminlu — APPROVED** [review]

File: src/main/java/wolox/training/security/CustomAuthenticationProvider.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    public CustomAuthenticationProvider() {}
+
+    @Override
+    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+        String name = authentication.getName();
+        String password = authentication.getCredentials().toString();
+        User user = userRepository.findByName(name);
+        if (user == null) {
+            throw new BadCredentialsException("Invalid user or password");
```
Comment (IgnOliveto) [inline src/main/java/wolox/training/security/CustomAuthenticationProvider.java]: Please add this error message to the  error message class.

File: src/main/java/wolox/training/repositories/UserRepository.java [hunk 2]
```diff
@@ -1,6 +1,10 @@
 package wolox.training.repositories;
 
+<<<<<<< HEAD
```
Comment (IgnOliveto) [inline src/main/java/wolox/training/repositories/UserRepository.java]: Please remove this merge conflict markers

**Review by IgnOliveto — COMMENTED** [review]
Some minor changes

**Pull request merged**